From dfe045f78f28a0c29ab9bb68a5e179f162e3fcdd Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Tue, 6 Nov 2018 15:01:35 +0800
Subject: [PATCH 5141/5242] MLK-20244 VPU: Config mem_printf base and size
 through structure DebugBufferDesc for vpu decoder

commit  c1e7697f08b206d6a0d8f8c87e02619b73e16229 from
https://source.codeaurora.org/external/imx/linux-imx.git

Config mem_printf base and size through structure DebugBufferDesc for
vpu decoder for debug on M0 core

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c  |    3 ---
 drivers/mxc/vpu-decoder-b0/vpu_b0.h  |    1 -
 drivers/mxc/vpu-decoder-b0/vpu_rpc.c |    4 ++--
 drivers/mxc/vpu-decoder-b0/vpu_rpc.h |    3 ++-
 4 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index c69e0da..aab700f 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -2128,7 +2128,6 @@ static irqreturn_t fsl_vpu_mu_isr(int irq, void *This)
 		rpc_init_shared_memory(&dev->shared_mem, dev->m0_rpc_phy - dev->m0_p_fw_space_phy, dev->m0_rpc_virt, SHARED_SIZE);
 		rpc_set_system_cfg_value(dev->shared_mem.pSharedInterface, VPU_REG_BASE);
 
-		MU_sendMesgToFW(dev->mu_base_virtaddr, PRINT_BUF_OFFSET, dev->m0_rpc_phy - dev->m0_p_fw_space_phy + M0_PRINT_OFFSET);
 		MU_sendMesgToFW(dev->mu_base_virtaddr, RPC_BUF_OFFSET, dev->m0_rpc_phy - dev->m0_p_fw_space_phy); //CM0 use relative address
 		MU_sendMesgToFW(dev->mu_base_virtaddr, BOOT_ADDRESS, dev->m0_p_fw_space_phy);
 		MU_sendMesgToFW(dev->mu_base_virtaddr, INIT_DONE, 2);
@@ -3273,8 +3272,6 @@ static int init_vpudev_parameters(struct vpu_dev *dev)
 
 	memset_io(dev->str_base_vir, 0, dev->str_size);
 #endif
-	rpc_init_shared_memory(&dev->shared_mem, dev->m0_rpc_phy - dev->m0_p_fw_space_phy, dev->m0_rpc_virt, SHARED_SIZE);
-	rpc_set_system_cfg_value(dev->shared_mem.pSharedInterface, VPU_REG_BASE);
 
 	return 0;
 }
diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.h b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
index f4a4d63..73ee889 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.h
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.h
@@ -78,7 +78,6 @@ struct vpu_v4l2_control {
 typedef enum{
 	INIT_DONE = 1,
 	RPC_BUF_OFFSET,
-	PRINT_BUF_OFFSET,
 	BOOT_ADDRESS,
 	COMMAND,
 	EVENT
diff --git a/drivers/mxc/vpu-decoder-b0/vpu_rpc.c b/drivers/mxc/vpu-decoder-b0/vpu_rpc.c
index d3a3b95..38f7f2b 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_rpc.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_rpc.c
@@ -133,12 +133,12 @@ void rpc_init_shared_memory(struct shared_addr *This,
 
 	phy_addr += QMETER_SIZE;
 	pDebugBufferDesc = &pSharedInterface->DebugBufferDesc;
-	pDebugBufferDesc->uWrPtr = phy_addr;
+	pDebugBufferDesc->uWrPtr = base_phy_addr + M0_PRINT_OFFSET;
 	pDebugBufferDesc->uRdPtr = pDebugBufferDesc->uWrPtr;
 	pDebugBufferDesc->uStart = pDebugBufferDesc->uWrPtr;
 	pDebugBufferDesc->uEnd = pDebugBufferDesc->uStart + DEBUG_SIZE;
 
-	phy_addr += DEBUG_SIZE;
+	phy_addr += sizeof(MediaIPFW_Video_BufDesc);
 	for (i = 0; i < VPU_MAX_NUM_STREAMS; i++) {
 		pEngAccessBufferDesc = &pSharedInterface->EngAccessBufferDesc[i];
 		pEngAccessBufferDesc->uWrPtr = phy_addr;
diff --git a/drivers/mxc/vpu-decoder-b0/vpu_rpc.h b/drivers/mxc/vpu-decoder-b0/vpu_rpc.h
index b777e58..d774dda 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_rpc.h
+++ b/drivers/mxc/vpu-decoder-b0/vpu_rpc.h
@@ -65,9 +65,10 @@
 #define GOP_SIZE 0x1000
 #define PIC_SIZE 0x1000
 #define QMETER_SIZE 0x1000
-#define DEBUG_SIZE 0x1000
+#define DEBUG_SIZE 0x200000
 #define ENG_SIZE 0x1000
 #define LOCAL_MSG_NUM VID_API_MESSAGE_LIMIT
+#define M0_PRINT_OFFSET 0x500000
 
 struct shared_addr {
 	pDEC_RPC_HOST_IFACE pSharedInterface;
-- 
1.7.9.5

